stellargraph/utils/saliency_maps/integrated_gradients.py

```diff
@@ -140,36 +193,58 @@ def get_integrated_link_masks(
         return total_gradients
 
     def get_node_importance(
-        self,
-        node_idx,
-        class_of_interest,
-        X_val=None,
-        A_index=None,
-        A_val=None,
-        steps=20,
+        self, node_idx, class_of_interest, steps=20,
     ):
         """
         The importance of the node is defined as the sum of all the feature importance of the node.
 
         Args:
-            Refer to the parameters in get_integrated_node_masks.
+            node_idx: the index of the node to calculate gradients for.
+            class_of_interest: the index for the class probability that the gradients will be calculated for.
+            steps (int): The number of values we need to interpolate. Generally steps = 20 should give good enough results.
 
         return (float): Importance score for the node.
         """
-        if X_val is None:
-            X_val = self.X
-        if A_index is None and self.is_sparse:
-            A_index = self.A_indices
-        if A_val is None:
-            A_val = self.A
 
         gradients = self.get_integrated_node_masks(
-            node_idx,
-            class_of_interest,
-            steps=steps,
-            X_val=X_val,
-            A_index=A_index,
-            A_val=A_val,
+            node_idx, class_of_interest, steps=steps,
         )
 
         return np.sum(gradients, axis=-1)
+
+    def _compute_gradients(self, mask_tensors, variable):
+
+        for i, x in enumerate(mask_tensors):
+            if not isinstance(x, tf.Tensor):
+                mask_tensors[i] = tf.convert_to_tensor(x)
+
+        if self.is_sparse:
+            (
+                features_t,
+                output_indices_t,
+                adj_indices_t,
+                adj_t,
+                class_of_interest,
+            ) = mask_tensors
+            model_input = [features_t, output_indices_t, adj_indices_t, adj_t]
+
+        else:
+            (features_t, output_indices_t, adj_t, class_of_interest,) = mask_tensors
+            model_input = [features_t, output_indices_t, adj_t]
+
+        with tf.GradientTape() as tape:
+            if variable == "nodes":
```

Comment:
what about an index for a variable in `model_inputs`? slightly more confusing but better reusability?

Reply:
Hmm yeah I guess that would make this method more generic.. like `w.r.t.` parameter that points to a variable in `model_inputs`? But I also agree it would be more confusing, could we even just pass in the same tensor twice..?
```python
grad = _compute_gradient(input=[features, adj, output], wrt=0)
# or
grad = _compute_gradient(input=[features, adj, output], wrt=features)
```
I'm not really sure that's better.

Reply:
I think that will work as long as we're passing in tensors not numpy arrays - I'll give it a go